gRPC code generation: support custom protoc location (#15264)
quarkusio/quarkus: docs/src/main/asciidoc/grpc-getting-started.adoc

With this configuration, you can put your service and message definitions in the `src/main/proto` directory.
`quarkus-maven-plugin` will generate Java files from your `proto` files.

> `quarkus-maven-plugin` obtains a version of `protoc` (protobuf compiler) that should suit your system from Maven repository.

Reviewer: ```suggestion
`quarkus-maven-plugin` retrieves a version of `protoc` (the protobuf compiler) from Maven repositories. The retrieved version matches your operating system and CPU architecture.```

> If this version does not work for you, you can download the suitable binary and specify its location via

Reviewer: ```suggestion
If this retrieved version does not work in your context, you can download the suitable binary and specify the location via
```

That's why we use the `os-maven-plugin` to target the executable compatible with the operating system.

NOTE: This configuration instructs the `protobuf-maven-plugin` to generate the default gRPC classes and classes using Mutiny to fit with the Quarkus development experience.

> IMPORTANT: When using `protobuf-maven-plugin`, every time you update the `proto` files, you need to re-generate the classes (using `mvn compile`).

Reviewer: ```suggestion
IMPORTANT: When using `protobuf-maven-plugin` instead of the `quarkus-maven-plugin`, every time you update the `proto` files, you need to re-generate the classes (using `mvn compile`).
```
